chapter_multilayer-perceptrons/dropout.md

```diff
@@ -423,10 +429,17 @@ class DropoutMLP(d2l.Classifier):
         x = nn.relu(nn.Dense(self.num_hiddens_2)(x))
         x = nn.Dropout(self.dropout_2, deterministic=not self.training)(x)
         return nn.Dense(self.num_outputs)(x)
+```
 
-@d2l.add_to_class(DropoutMLP)
-def loss(self, params, X, Y, averaged=True, rngs={'dropout': d2l.get_key()}):
-    return super(DropoutMLP, self).loss(params, X, Y, averaged, rngs)
+```{.python .input}
+%%tab jax
+@d2l.add_to_class(d2l.Classifier)  #@save
+@partial(jax.jit, static_argnums=(0,4))
+def loss(self, params, X, Y, averaged=True):
```

Comment:
You seem to overwrite Classifier.loss. 
Do we still need to keep the `averaged` argument?

Reply:
Yes, as mentioned in the note the only difference is that we pass a specific rng for dropout which helps flax to generate the dropout mask. This is required in training mode for flax during `model.apply()`